main/decorators.py

```diff
@@ -0,0 +1,12 @@
+from django.shortcuts import redirect
```

Comment:
Does this built in django decorator do what you're trying to do here?
https://docs.djangoproject.com/en/3.2/topics/auth/default/#the-login-required-decorator

Reply:
I wrote the opposite decorator, if you're logged in, you can't access the page